### PR TITLE
Load plugins explicitly in the code

### PR DESCRIPTION
**Pull request by mfn**
The current implicit approach has some mental overhead, as first once
has to discover the `require_once` directives _and then_ realize
each of the extension has its own auto-register code at the end.

Explicitly adding them makes them easier to reason about and the
dedicated method allows to even alter the default plugins, if desired.

Also fixes the filename of the `ExtensionInterface` so the autoload
correctly picks it up.

### CODE REVIEW

File: src/Extension/Choice.php [hunk 1]
```diff
@@ -53,5 +53,3 @@ private function buildInternalSequence(&$grammar, $sequence, $grammarParser, $op
         return (count($choice) === 1) ? $choice[0] : $choice;
     }
 }
-
-\ParserGenerator\GrammarParser::$defaultPlugins[] = new Choice();
```
Comment (mfn) [inline src/Extension/Choice.php:57]: All of those lines have been moved to `\ParserGenerator\GrammarParser::addDefaultPlugins`

File: src/Extension/ItemRestrictions.php [hunk 2]
```diff
@@ -2,13 +2,6 @@
 
 namespace ParserGenerator\Extension;
 
-require_once('ItemRestrictions/ItemRestrictionInterface.php');
-require_once('ItemRestrictions/ItemRestrictionOr.php');
-require_once('ItemRestrictions/ItemRestrictionAnd.php');
-require_once('ItemRestrictions/ItemRestrictionNot.php');
-require_once('ItemRestrictions/Contain.php');
-require_once('ItemRestrictions/Is.php');
```
Comment (mfn) [inline src/Extension/ItemRestrictions.php:10]: None of them are necessary, those get correctly autoloaded

File: src/GrammarParser.php [hunk 3]
```diff
@@ -13,6 +27,31 @@ class GrammarParser
...
 
     public function __construct()
+    {
+        $this->addDefaultPlugins();
+        $this->loadPlugins();
+    }
+
+    public function addDefaultPlugins()
+    {
+        // Note: load order does matter
```
Comment (mfn) [inline src/GrammarParser.php:37]: At first I sorted them alphabetically but then the test `\ParserGenerator\Tests\Extension\LookaheadTest::testErrorTrack` failed: `TextNode` needed to be loaded before `Lookahead`.

I then decided to simply keep the original order from the `require_once` directives from the end of the file.

**Comment by farafiri** [comment]
Yep. I totally forgot about these requires. I started developing this project before composer with its autoloader was widespread solution.

**Pull request merged**